linode/networkingip/framework_datasource.go

```diff
@@ -74,10 +70,29 @@ func (d *DataSource) Read(
 
 	ctx = tflog.SetField(ctx, "address", data.Address.ValueString())
 
-	ip, err := d.Meta.Client.GetIPAddress(ctx, data.Address.ValueString())
+	// Workaround: Use GetIPAddresses and filter for the specific address
+	ips, err := d.Meta.Client.ListIPAddresses(ctx, nil)
 	if err != nil {
 		resp.Diagnostics.AddError(
-			"Unable to get IP Address: %s", err.Error(),
+			"Unable to list IP Addresses",
+			err.Error(),
+		)
+		return
+	}
+
+	var ip *linodego.InstanceIP
+	for _, candidate := range ips {
+		if candidate.Address == data.Address.ValueString() {
```

Comment:
It is possible that the same IP address is present but from different regions for a private address. Can the region is be checked as well?

Reply:
We could additionally filter the results of the ListIPAddresses() based on the region. Will be making the change!

Reply:
FYI, the region is needed to disambiguate only if it is an RFC1918 address and there is a duplicate